stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -757,6 +670,190 @@ impl NakamotoBlockProposal {
+                                // block limit reached; add tx back to replay set.
+                                // BUT we know that the block should have ended at this point, so
+                                // return an error.
+                                let txid = replay_tx.txid();
+                                replay_txs.push_front(replay_tx);
+
+                                warn!("Rejecting block proposal. Next replay tx exceeds cost limits, so should have been in the next block.";
+                                    "error" => %error,
+                                    "txid" => %txid,
+                                );
+
+                                return Err(BlockValidateRejectReason {
+                                    reason_code: ValidateRejectCode::InvalidTransactionReplay,
+                                    reason: "Transaction is not in the replay set".into(),
+                                });
+                            }
+                            _ => {
+                                info!("During replay block validation, allowing problematic tx to be dropped";
+                                    "txid" => %replay_tx.txid(),
+                                    "error" => %error,
+                                );
+                                // it's ok, drop it
+                                continue;
+                            }
+                        }
+                    }
+                    TransactionResult::Success(_) => {
+                        // Tx should have been included
+                        warn!("Rejected block proposal. Block doesn't contain replay transaction that should have been included.";
+                            "block_txid" => %tx.txid(),
+                            "block_tx_index" => i,
+                            "replay_txid" => %replay_tx.txid(),
+                        );
+                        return Err(BlockValidateRejectReason {
+                            reason_code: ValidateRejectCode::InvalidTransactionReplay,
+                            reason: "Transaction is not in the replay set".into(),
+                        });
+                    }
+                };
+            }
+
+            // Apply the block's transaction to our block builder, but we don't
+            // actually care about the result - that happens in the main
+            // validation check.
+            let _tx_result = replay_builder.try_mine_tx_with_len(
+                &mut replay_tenure_tx,
+                tx,
+                tx_len,
+                &BlockLimitFunction::NO_LIMIT_HIT,
+                ASTRules::PrecheckSize,
+                None,
+            );
+        }
+
+        let no_replay_txs_remaining = replay_txs.is_empty();
+
+        // Now, we need to check if the remaining replay transactions are unmineable.
+        let only_unmineable_remaining = replay_txs.is_empty()
+            || replay_txs.iter().all(|tx| {
```

Comment:
Currently, `only_unmineable_remaining` behaves the same as the condition `no_replay_txs_remaining || only_unmineable_remaining` (even if we remove the `replay_txs.is_empty()` check from its definition).
Because of this, the final result `Ok(no_replay_txs_remaining || only_unmineable_remaining)` is effectively redundant or "superseded" by `only_unmineable_remaining`.

If we want to keep the final result for the sake of clarity or expressiveness, we should redefine `only_unmineable_remaining` to reflect only the case where non-empty replay transactions are all unmineable, like so:
```suggestion
        let no_replay_txs_remaining = replay_txs.is_empty();

        // Now, we need to check if the remaining replay transactions are unmineable.
        let only_unmineable_remaining = !replay_txs.is_empty()
            && replay_txs.iter().all(|tx| {
```

Alternatively, we could merge both conditions into a single boolean with a more descriptive (albeit possibly long) name, such as `no_replay_txs_or_only_unmineable_remaining`.